### ISSUE TITLE
Support `to` in uncertainty

### ISSUE THREAD

**Issue by eteq** [astropy/astropy#8863]
Prompted by a question from @parejkoj, I noticed the current behavior 
```
>>> from astropy import uncertainty as unc
>>> from astropy import units as u
>>> flux = unc.normal(100*u.AB, std=1*u.AB, n_samples=10000)
>>> flux.to(u.MagUnit(u.AB))
ValueError: Value not scalar compatible or convertible to an int, float, or complex array
```

It really would be better if that worked as expected, because that's basically all that's needed for magnitude/flux error propagation using the canonical machinery in astropy for this.

The broader problem seems to be that *any* `to` method-call yields this same error though, so I think this may be a bug...?

**Comment by astrofrog** [comment]
Also we should make sure ``to_value`` works

**Comment by privong** [comment]
For what it's worth, in my informal checking of this it does seem that `to_value()` works for the cases I've tried. There's an example in #9055.

**Comment by astrofrog** [comment]
Ran into this again today - all unit conversions fail! This is a pretty annoying bug and I think we should definitely fix for 4.0!

**Comment by pllim** [comment]
You have 3 days...

**Comment by mhvk** [comment]
fixed by #9429 and #9442
